service/history/shard/context_impl.go

```diff
@@ -193,6 +195,14 @@ func (s *ContextImpl) GetMaxTaskIDForCurrentRangeID() int64 {
 	return s.maxTaskSequenceNumber - 1
 }
 
+func (s *ContextImpl) GetVClock() *clockpb.ShardClock {
+	s.rLock()
+	clock := s.taskSequenceNumber
```

Comment:
Is it guaranteed that any update to any workflow (in the shard) will cause this to increment? For example, if there is already a workflow task scheduled (but not started), and there comes a new signal, will the new signal cause the taskSequenceNumber to increase?

Reply:
Maybe that doesn't matter since this is mainly to detect stale shard (which happen only if someone steal it) in that case, the clock must have moved forward.

Reply:
>Is it guaranteed that any update to any workflow (in the shard) will cause this to increment? For example, if there is already a workflow task scheduled (but not started), and there comes a new signal, will the new signal cause the taskSequenceNumber to increase?

yes, newly generated history event will use this sequence number

this `GetVClock` is a read method, if business logic need to stamp an event, then there should be a generate method